bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/HttpServiceUtil.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.net;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+
+/**
+ * Some utility functions related to the http service
+ *
+ * @author Markus Rathgeb - Initial contribution and API
+ */
+public class HttpServiceUtil {
+
+    private HttpServiceUtil() {
+    }
+
+    /**
+     * Get the port that is used by the HTTP service.
+     *
+     * @param bc the bundle context used for lookup
+     * @return the port if used, -1 if no port could be found
+     */
+    public static int getHttpServicePort(final BundleContext bc) {
+        return getHttpServicePortProperty(bc, "org.osgi.service.http.port");
+    }
+
+    public static int getHttpServicePortSecure(final BundleContext bc) {
+        return getHttpServicePortProperty(bc, "org.osgi.service.http.port.secure");
+    }
+
+    // Utility class that could be used for non-secure and secure port.
+    private static int getHttpServicePortProperty(final BundleContext bc, final String propertyName) {
+        Object value;
+        int port = -1;
+
+        // Try to find the port by using the service property (respect service ranking).
+        try {
+            int candidate = Integer.MIN_VALUE;
+            final ServiceReference[] refs = bc.getAllServiceReferences("org.osgi.service.http.HttpService", null);
+            for (final ServiceReference ref : refs) {
+                value = ref.getProperty(propertyName);
+                if (value == null) {
+                    continue;
+                }
+                final int servicePort;
+                try {
+                    servicePort = Integer.parseInt(value.toString());
+                } catch (final NumberFormatException ex) {
+                    continue;
+                }
+                value = ref.getProperty(Constants.SERVICE_RANKING);
+                final int serviceRanking;
+                if (value == null || !(value instanceof Integer)) {
+                    serviceRanking = 0;
+                } else {
+                    serviceRanking = (Integer) value;
+                }
+                if (serviceRanking >= candidate) {
+                    candidate = serviceRanking;
+                    port = servicePort;
+                }
+            }
+        } catch (final InvalidSyntaxException ex) {
```

Comment:
You are right, a log is normally okay.
But using a log in the static method involves the creation of a static logger -- but okay, this could be done, too.

If you have a look at the API documentation the `InvalidSyntaxException` is thrown "If the specified filter contains an invalid filter expression that cannot be parsed."
filter is null and so not used :wink: 
Perhaps a line comment in the code would be nice to explain why the exception is ignored silently.